File: Programmers/42840/42840.py
def solution(answers):
    answer = []
    first = [1,2,3,4,5]
    second = [2,1,2,3,2,4,2,5]
    third = [3,3,1,1,2,2,4,4,5,5]
    tmp = [0, 0, 0]

    for index in range(len(answers)):
        if answers[index] == first[index % 5]:
            tmp[0] += 1
        if answers[index] == second[index % 8]:
            tmp[1] += 1
        if answers[index] == third[index % 10]:
            tmp[2] += 1

    for index in range(len(tmp)):
        if tmp[index] == max(tmp):
            answer.append(index + 1)
    return answer

File: Programmers/42840/42840.go
package moketest

func solution(answers []int) []int {
	answer := []int{}

	pick1 := [...]int{1, 2, 3, 4, 5}
	pick2 := [...]int{2, 1, 2, 3, 2, 4, 2, 5}
	pick3 := [...]int{3, 3, 1, 1, 2, 2, 4, 4, 5, 5}

	correct := [3]int{0, 0, 0}

	//정답 체크
	for i, v := range answers {
		if v == pick1[i%len(pick1)] {
			correct[0]++
		}
		if v == pick2[i%len(pick2)] {
			correct[1]++
		}
		if v == pick3[i%len(pick3)] {
			correct[2]++
		}
	}
	m := 0
	for i, e := range correct {
		if i == 0 || e > m {
			m = e
		}
	}
	for i, e := range correct {
		if e == m {
			answer = append(answer, i+1)
		}
	}
	return answer
}
